### ISSUE TITLE
Sticky header goes below floating SliverAppBar()

### ISSUE THREAD

**Issue by emvaized** [fluttercommunity/flutter_sticky_headers#33]
When SliverAppBar inside CustomScrollView is set to be shown only on scroll up (floating: true), header stays on its place e.g. goes below appbar. How to make it move down on appbar showing?

P.S.: I don't want to use this [flutter_sticky_header](https://pub.dev/packages/flutter_sticky_header), since it can't work with ListView builder.

**Comment by tp7309** [comment]
You can try [sticky_and_expandable_list](https://pub.dev/packages/sticky_and_expandable_list), just use as one SliverList.
![example](https://raw.githubusercontent.com/tp7309/flutter_sticky_and_expandable_list/master/doc/images/sliverlist.gif)

**Comment by emvaized** [comment]
@tp7309 

It works, but for some reason sticky headers now go below statusbar in collapsed mode (code from example).

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/37851576/76683962-b3300500-6610-11ea-96f3-eebd71f5ca97.gif)

**Comment by tp7309** [comment]
> @tp7309
> 
> It works, but for some reason sticky headers now go below statusbar in collapsed mode (code from example_sliver.dart).

The hidden height is statusbar height, I have updated the example project.
Just wrap *Scaffold* with **SafeArea**.

**Comment by emvaized** [comment]
@tp7309 
Great, bug is fixed now! Thanks!

My only question remaining is -- how to make headers and content share the same index in list? 
In my case of usage each section contains only one header and one child, and I want them to share the same index.

Current behavior:
![Screenshot_20200315_153047](https://user-images.githubusercontent.com/37851576/76702371-f900d200-66d1-11ea-8e7c-deea34c40f80.png)

Desired:
![Screenshot_20200315_153047](https://user-images.githubusercontent.com/37851576/76702508-0c606d00-66d3-11ea-81a9-18c5f5bc82ca.png)

**Comment by tp7309** [comment]
> @tp7309
> Great, bug is fixed now! Thanks!
> 
> My only question remaining is -- how to make headers and content share the same index in list?
> In my case of usage each section contains only one header and one child, and I want them to share the same index.

Echo itemBuilder return the **section** model and **item** model, you can get section/header index from section model, example:
```flutter
            SliverExpandableList(
              builder: SliverExpandableChildDelegate<String, Section>(
                  sectionList: sectionList,
                  headerBuilder: _buildHeader,
                  itemBuilder: (context, section, item, index) {
                    //get sectionIndex index from sectionList index
                    int sectionIndex = sectionList.indexOf(section);
                    return ListTile(
                      leading: CircleAvatar(
                        child: Text("$sectionIndex"),
                      ),
                      title: Text(item),
                    );
                  }),
            )
```

**Comment by emvaized** [comment]
@tp7309 
Works great now, thanks a lot! 
Your package is a great alternative for the two already existing 'sticky-headers' packages.

Some ideas for improving:
1) Built-in collapsing animation for `setSectionExpanded()` method;
2) Ability to integrate with `LiveSliverList` from [auto_animated](https://pub.dev/packages/auto_animated) package;
3) Make `Section()` constructor not only String-oriented, but to accept widgets as well.

**Comment by tp7309** [comment]
@emvaized 
Good idea! Animation support coming soon, you can star repo for receive lastest version update.
Some explanations about [Option 3](https://github.com/tp7309/flutter_sticky_and_expandable_list/issues/1).

**Comment by emvaized** [comment]
> @emvaized
> Good idea! Animation support coming soon, you can star repo for receive lastest version update.
> Some explanations about [Option 3](https://github.com/tp7309/flutter_sticky_and_expandable_list/issues/1).
@tp7309 
Cool, thanks for your work! I'll try it.

Also I wanted to ask you about this error, which occurs on every launch of the app, even if there's no `SliverExpandableList` widget built on the screen.
![Screenshot_20200316_181606](https://user-images.githubusercontent.com/37851576/76778399-713ec480-67b2-11ea-8955-18fd6ec32e2d.png)

**Comment by emvaized** [comment]
However, I will leave this issue open, since the problem still exists in discussed package.

@slightfoot Is [flutter_sticky_headers](https://github.com/fluttercommunity/flutter_sticky_headers) abandoned?

**Comment by sohailxi** [comment]
> @tp7309
> 
> It works, but for some reason sticky headers now go below statusbar in collapsed mode (code from example_sliver.dart).
> 
> ![ezgif com-video-to-gif](https://user-images.githubusercontent.com/37851576/76683962-b3300500-6610-11ea-96f3-eebd71f5ca97.gif) [ ![ezgif com-video-to-gif](https://user-images.githubusercontent.com/37851576/76683962-b3300500-6610-11ea-96f3-eebd71f5ca97.gif) ](https://user-images.githubusercontent.com/37851576/76683962-b3300500-6610-11ea-96f3-eebd71f5ca97.gif) [ ](https://user-images.githubusercontent.com/37851576/76683962-b3300500-6610-11ea-96f3-eebd71f5ca97.gif)

How did you solve this I am facing the same problem can you share the exact solution with me 

Uploading VID_20230118_200940.mp4…

**Comment by tp7309** [comment]
> @kanchasohail 

you need try or open an issue in [flutter_sticky_and_expandable_list](https://github.com/tp7309/flutter_sticky_and_expandable_list)
